examples/Microsoft.Spark.CSharp.Examples/MachineLearning/SentimentAnalysis.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+//using Microsoft.ML;
+//using Microsoft.ML.Data;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.Examples.MachineLearning
+{
+    /// <summary>
+    /// Example of using ML.NET + .NET for Apache Spark
+    /// for sentiment analysis.
+    /// </summary>
+    internal sealed class SentimentAnalysis : IExample
+    {
+        public void Run(string[] args)
+        {
+            // Change this flag once you've set up
+            // the ML.NET dependencies described in the README
+            int dependenciesDone = 0;
+
+            if (args.Length != 1)
+            {
+                Console.Error.WriteLine(
+                    "Usage: SentimentAnalysis <path to review dataset>");
+
+                Environment.Exit(1);
+            }
+
+            if (dependenciesDone == 0)
+            {
+                Console.Error.WriteLine(
+                    "You need to complete setting ML.NET dependencies");
+
+                Environment.Exit(1);
+            }
+
+            SparkSession spark = SparkSession
+                .Builder()
+                .AppName(".NET for Apache Spark Sentiment Analysis")
+                .GetOrCreate();
+
+            DataFrame df = spark
+                .Read()
+                .Option("header", true)
+                .Option("inferSchema", true)
+                .Csv(args[0]);
+            df.Show();
+
+            // Use ML.NET to evaluate each review 
+            /*spark.Udf().Register<string, bool>(
```

Comment:
Reason why this code block commented out on purpose ?

Reply:
I could definitely use your team's advice for how to handle this. 

I wanted to include the ML.NET apps in our examples folder, but if I left the app in there as-is (no commented out sections/blocks), when I went to run the examples, I would get errors that none of the ML.NET libraries could be found. I think finding a way to install/include the ML.NET nuget could address this? Or we'd have to include the ML.NET dlls?

Reply:
For instance, when I run the publish command within the Microsoft.Spark.CSharp.Examples folder, if I leave the Microsoft.ML using statements uncommented, I get the error:

MachineLearning\SentimentAnalysis.cs(7,17): error CS0234: The type or namespace name 'ML' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?) [C:\Users\brmurtau\Documents\GitHub\sparkedits\examples\Microsoft.Spark.CSharp.Examples\Microsoft.Spark.CSharp.Examples.csproj]
MachineLearning\SentimentAnalysis.cs(8,17): error CS0234: The type or namespace name 'ML' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?) [C:\Users\brmurtau\Documents\GitHub\sparkedits\examples\Microsoft.Spark.CSharp.Examples\Microsoft.Spark.CSharp.Examples.csproj]

Reply:
Have you tried modifying [Microsoft.Spark.CSharp.Examples.csproj](https://github.com/dotnet/spark/blob/master/examples/Microsoft.Spark.CSharp.Examples/Microsoft.Spark.CSharp.Examples.csproj) and adding the following ?

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.ML" Version="1.3.1" />
  </ItemGroup>
```

Reply:
So I just tried adding that, and that fix will now let it compile without needing to comment out ML.NET code! 

Granted, the user also has to copy over all of the Microsoft.ML dlls in order to run a successful `spark-submit` command (or else get errors like: `[2019-10-22T17:13:15.1682561Z] [MININT-M0C5UII] [Error] [TaskRunner] [0] ProcessStream() failed with exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.ML.FastTree, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.`). 

But since all the examples will successfully compile without having these blocks commented out, I will uncomment them and ensure the ML apps have sufficient readme's explaining the need to copy over dll's for spark-submit.